lib/grammers-client/src/client/files.rs

```diff
@@ -231,35 +232,44 @@ impl Client {
 
         if big_file {
             let parts = Arc::new(parts);
-            let mut tasks = Vec::with_capacity(WORKER_COUNT);
-            for _ in 0..WORKER_COUNT {
-                let handle = self.clone();
+            let mut connections = Vec::with_capacity(CONNECTION_COUNT);
+            for _ in 0..CONNECTION_COUNT {
                 let parts = Arc::clone(&parts);
-                let task = async move {
-                    while let Some((part, bytes)) = parts.next_part().await? {
-                        let ok = handle
-                            .invoke(&tl::functions::upload::SaveBigFilePart {
-                                file_id,
-                                file_part: part,
-                                file_total_parts: total_parts,
-                                bytes,
-                            })
-                            .await
-                            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-
-                        if !ok {
-                            return Err(io::Error::new(
-                                io::ErrorKind::Other,
-                                "server failed to store uploaded data",
-                            ));
-                        }
+                let connection = async move {
+                    let mut tasks = Vec::with_capacity(WORKER_COUNT);
+                    let client = Client::connect(self.0.config.clone()).await.unwrap();
```

Comment:
I need to look deeper into this, but I'm not sure creating an entire new client is the route we want to go. (The `unwrap()` definitely needs to go though.)

This also introduces an ever greater chance for things to go wrong (what if one connection fails but the others don't?).

Reply:
We could return an IO error instead, the try join method will fail immediately if any of the clients failed to connect.

Reply:
Would it not be better to fallback to a single connection approach? Or however many workers managed to connect.

Also, this is not a pool. If my bot receives a command from a hundred users at once and I need to serve them all at the same time, this would create 300 connections.